staging/src/k8s.io/dynamic-resource-allocation/deviceattribute/sysfs_linux.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deviceattribute
+
+import "path/filepath"
+
+const (
+	sysfsRoot = "/sys"
+)
+
+var (
+	sysfs sysfsPath = sysfsPath(sysfsRoot)
+)
+
+// sysfsPath provides methods to construct sysfs paths for various subsystems.
+// It is used to abstract the sysfs path construction
+// and can be replaced with a mock in tests.
+type sysfsPath string
+
+func (s sysfsPath) Devices(path string) string {
```

Comment:
```suggestion
func (s sysfsPath) devices(path string) string {
```

Same below.

In this case, exporting the method makes it callable outside of the package (assuming that some exported method returns a sysfsPath) to that caller and thus part of the package API. We want a lean API to make future changes safer.